src/devSVG.cpp

```diff
@@ -241,55 +290,59 @@ void svg_clip(double x0, double x1, double y0, double y1, pDevDesc dd) {
 }
 
 void svg_new_page(const pGEcontext gc, pDevDesc dd) {
+BEGIN_RCPP
```

Comment:
Yeah if we call `Rf_error` directly we will indeed skip C++ destructors. This approach allows us to catch the C++ exception and then convert it to an `Rf_error` for the R C layers benefit.